gensim/models/poincare.py

```diff
@@ -731,27 +737,389 @@ def compute_loss(self):
         self._loss_computed = True
 
 
-class PoincareKeyedVectors(KeyedVectors):
+class PoincareKeyedVectors(KeyedVectorsBase):
     """Class to contain vectors and vocab for the :class:`~gensim.models.poincare.PoincareModel` training class.
 
     Used to perform operations on the vectors such as vector lookup, distance etc.
 
     """
+
+    def __init__(self):
+        super(PoincareKeyedVectors, self).__init__()
+        self.max_distance = 0
+
     @staticmethod
-    def poincare_dist(vector_1, vector_2):
-        """Return poincare distance between two vectors."""
-        norm_1 = np.linalg.norm(vector_1)
-        norm_2 = np.linalg.norm(vector_2)
-        euclidean_dist = np.linalg.norm(vector_1 - vector_2)
-        if euclidean_dist == 0.0:
-            return 0.0
-        else:
-            return np.arccosh(
-                1 + 2 * (
-                    (euclidean_dist ** 2) / ((1 - norm_1 ** 2) * (1 - norm_2 ** 2))
-                )
+    def vector_distance(vector_1, vector_2):
+        """
+        Return poincare distance between two input vectors. Convenience method over `vector_distance_batch`.
+
+        Parameters
+        ----------
+        vector_1 : numpy.array
+            input vector
+        vector_2 : numpy.array
+            input vector
+
+        Returns
+        -------
+        numpy.float
+            Contains Poincare distance between `vector_1` and `vector_2`.
+
+        """
+        return PoincareKeyedVectors.vector_distance_batch(vector_1, vector_2[np.newaxis, :])[0]
+
+    @staticmethod
+    def vector_distance_batch(vector_1, vectors_all):
+        """
+        Return poincare distances between one vector and a set of other vectors.
+
+        Parameters
+        ----------
+        vector_1 : numpy.array
+            vector from which Poincare distances are to be computed.
+            expected shape (dim,)
+        vectors_all : numpy.array
+            for each row in vectors_all, distance from vector_1 is computed.
+            expected shape (num_vectors, dim)
+
+        Returns
+        -------
+        numpy.array
+            Contains Poincare distance between vector_1 and each row in vectors_all.
+            shape (num_vectors,)
+
+        """
+        euclidean_dists = np.linalg.norm(vector_1 - vectors_all, axis=1)
+        norm = np.linalg.norm(vector_1)
+        all_norms = np.linalg.norm(vectors_all, axis=1)
+        return np.arccosh(
+            1 + 2 * (
+                (euclidean_dists ** 2) / ((1 - norm ** 2) * (1 - all_norms ** 2))
             )
-    # TODO: Add other KeyedVector supported methods - most_similar, etc.
+        )
+
+    def closest_child(self, node):
+        """
+        Returns the node closest to `node` that is lower in the hierarchy than `node`.
+
+        Parameters
+        ----------
+        node : str
```

Comment:
This always `str` (or int possible too)? This question more global (about all methods) that pass `node` argument?

Reply:
It could be an `int` too in theory, depending on what the vocab keys are. The most common case is `str` though. How would you prefer to handle this?

Reply:
maybe `str` or `int` everywhere?

Reply:
Sounds good

Reply:
Done. Also made some other changes to docstrings for more clarity.